### PR TITLE
lighttpd: refresh configs with upstream

### PR DESCRIPTION
**Pull request by gstrauss**
Topic Description
-----------------

* use pcre2 instead of pcre
* remove deprecated config options
* simplify lighttpd.conf (modern lighttpd has a built-in mimetype.assign with commonly used web file types)
* update systemd lighttpd.service to match upstream recommendations

Note: not modified: if `AUTOTOOLS_AFTER__RETRO` includes `--with-kerberos5`, then `PKGDEP__RETRO` should be updated to add a dependency on the appropriate package for krb5 libs.

Note: I did not understand the issue in #7296, so that is not addressed here.

Package(s) Affected
-------------------

lighttpd

Build Order
------------------

```
#buildit lighttpd
```

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [ ] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

Architectural progress for secondary ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Review by MingcongBai — DISMISSED** [review]
Sorry for the late review - it's been extremely busy lately. Can you please append:

```
REL=1
```

To the end of `app-network/lighttpd/spec` to indicate a revision?

**Comment by gstrauss** [comment]
rebased to tip of 'stable' branch and REL=1 added to app-network/lighttpd/spec, as requested.

**Review by KexyBiscuit — COMMENTED** [review]
It builds, it works, code looks good, thank you for your contribution! Waiting on approval from @MingcongBai for now.

```
kexybiscuit@ProArtB550-CREATOR [ ~ ] $ dpkg -s lighttpd
Package: lighttpd
Status: install ok installed
Section: net
Installed-Size: 1368
Maintainer: AOSC OS Maintainers <maintainers@aosc.io>
Architecture: amd64
Version: 1.4.76-1~pre20240814T170425Z
Depends: libxml2 (>= 2.10.4-1), lua (>= 1:5.1.5-5), mariadb (>= 10.9.8), openldap (>= 2.6.7), pcre2 (>= 10.44), sqlite (>= 3.45.1), systemd (>= 1:256.4), util-linux (>= 2.40.1)
Conffiles:
 /etc/lighttpd/lighttpd.conf c1b7b7ecdcf0b3046c2ed122c7265f93
 /etc/logrotate.d/lighttpd 8ab06aec207475c951996a8cbdd4b543
Description: A secure, fast, compliant and very flexible web server
X-AOSC-ACBS-Version: 20240808
X-AOSC-Commit: b0d056e64dX-AOSC-Packager: AOSC OS Maintainers <maintainers@aosc.io>
X-AOSC-Autobuild4-Version: 4.3.19
kexybiscuit@ProArtB550-CREATOR [ ~ ] $ sudo systemctl start lighttpd
kexybiscuit@ProArtB550-CREATOR [ ~ ] $ systemctl status lighttpd
● lighttpd.service - Lighttpd Web Server
     Loaded: loaded (/usr/lib/systemd/system/lighttpd.service; disabled; preset: disabled)
     Active: active (running) since Thu 2024-08-15 01:37:31 CST; 2s ago
 Invocation: e5a3eabc2374457ab205e8f67d7a5588
    Process: 1277 ExecStartPre=/usr/sbin/lighttpd -tt -f /etc/lighttpd/lighttpd.conf (code=exited, status=0/SUCCESS)
   Main PID: 1279 (lighttpd)
      Tasks: 1 (limit: 77088)
     Memory: 1.5M
     CGroup: /system.slice/lighttpd.service
             └─1279 /usr/sbin/lighttpd -D -f /etc/lighttpd/lighttpd.conf

Aug 15 01:37:31 ProArtB550-CREATOR systemd[1]: Starting lighttpd.service - Lighttpd Web Server...
Aug 15 01:37:31 ProArtB550-CREATOR systemd[1]: Started lighttpd.service - Lighttpd Web Server.
kexybiscuit@ProArtB550-CREATOR [ ~ ] $ curl -Iv localhost
* Host localhost:80 was resolved.
* IPv6: ::1
* IPv4: 127.0.0.1
*   Trying [::1]:80...
* connect to ::1 port 80 from ::1 port 43142 failed: Connection refused
*   Trying 127.0.0.1:80...
* Connected to localhost (127.0.0.1) port 80
> HEAD / HTTP/1.1
> Host: localhost
> User-Agent: curl/8.9.1
> Accept: */*
>
* Request completely sent off
< HTTP/1.1 404 Not Found
HTTP/1.1 404 Not Found
< Content-Type: text/html
Content-Type: text/html
< Content-Length: 158
Content-Length: 158
< Date: Wed, 14 Aug 2024 17:37:42 GMT
Date: Wed, 14 Aug 2024 17:37:42 GMT
< Server: lighttpd/1.4.76
Server: lighttpd/1.4.76
<

* Connection #0 to host localhost left intact
```

File: app-network/lighttpd/autobuild/overrides/usr/lib/systemd/system/lighttpd.service
Comment (KexyBiscuit) [inline app-network/lighttpd/autobuild/overrides/usr/lib/systemd/system/lighttpd.service]: We have `/usr/bin` and `/usr/sbin` merged in AOSC OS, so it should be like...

```suggestion
ExecStartPre=/usr/bin/lighttpd -tt -f /etc/lighttpd/lighttpd.conf
ExecStart=/usr/bin/lighttpd -D -f /etc/lighttpd/lighttpd.conf
```

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]
Just one small tip, everything else LGTM.

**Review by KexyBiscuit — APPROVED** [review]

**Comment by KexyBiscuit** [comment]
Merged and shipped, thank you!

Loop @MingcongBai for new contributor membership onboarding.

**Comment by MingcongBai** [comment]
> Merged and shipped, thank you!
> 
> Loop @MingcongBai for new contributor membership onboarding.

En route.

**Comment by MingcongBai** [comment]
@gstrauss You are entitled to an @aosc.io email address, a personal page on our home page, and access to our automation and build servers. I'm sending you the details via email.

**Pull request merged**
